### PR TITLE
JSDocs comments and initial design guidelines

### PR DESCRIPTION
**Pull request by blackbaud-johnly**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720](https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720)

File: src/app/public/modules/toast/toast.service.ts [hunk 1]
```diff
@@ -85,9 +85,10 @@ export class SkyToastService implements OnDestroy {
 
   /**
    * Opens a new toast using a custom component.
-   * @param component Angular component to inject into the toast body
-   * @param config Optional configuration
-   * @param providers Optional providers for the custom component
+   * @param component Specifies an Angular component to inject into the toast body,
+   * @param config Specifies additional configuration options for the toast.
+   * @param providers Specifies an array of custom providers to pass to the custom component's
+   * constructor. Returns a `SkyToastInstance`.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/toast/toast.service.ts]: Should "Returns a `SkyToastInstance`." be here in this parameter or in the method description? Seems like I copied this over to the wrong spot and it should be moved up.

File: src/app/public/modules/toast/toast.service.ts [hunk 2]
```diff
@@ -64,9 +64,9 @@ export class SkyToastService implements OnDestroy {
   }
 
   /**
-   * Opens a new toast with a text message.
-   * @param message Text to display inside the toast
-   * @param config Optional configuration
+   * Opens a new toast and displays the specified message.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/toast/toast.service.ts:67]: Should we have "Returns a `SkyToastInstance`." at the end here?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720](https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720)

File: src/app/public/modules/toast/toast-instance.ts [hunk 3]
```diff
@@ -13,6 +13,9 @@ export class SkyToastInstance {
     return this._closed;
   }
 
+  /**
+   * An observable that indicates when the toast is closed.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/toast/toast-instance.ts]: We have a section for `SkyToastInstance` properties in the current docs at https://developer.blackbaud.com/skyux/components/toast#skytoastinstance-properties, but the preview here doesn't generate that table. Do we need to do something to make it show up?

**Review by Blackbaud-DeniseCisneros — DISMISSED** [review]

File: src/app/public/modules/toast/toast.service.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast.service.ts]: You can remove that sentence altogether; it should infer the return type automatically.

File: src/app/public/modules/toast/toast.service.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast.service.ts:67]: Same as above.

File: src/app/public/modules/toast/toast-instance.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast-instance.ts]: Good catch. We have an issue filed for this, here: https://github.com/blackbaud/skyux-docs-tools/issues/30. I'll work on a fix for this ASAP.

File: src/app/public/modules/toast/toast.service.ts (hunk 1 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/toast/toast.service.ts]: Done. Thanks!

File: src/app/public/modules/toast/toast-instance.ts (hunk 3 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/toast/toast-instance.ts]: @Blackbaud-SteveBrush, given that we have an issue for this and I have the JSDocs comments in place, is there anything else to do here in this PR? Can we just merge this PR and wait for the table to be automatically generated when the `docs-tools` issue is resolved? Or do we need to hold off on merging? Or merge and then follow up later with a manual step after that `docs-tools` issue is addressed?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720](https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720)

File: src/app/public/modules/toast/toast-instance.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast-instance.ts]: Upgrading `@skyux/docs-tools` should fix this issue.

Comment (blackbaud-johnly) [reply src/app/public/modules/toast/toast-instance.ts]: I upgraded the `docs-tools` version, and that created the missing table for `SkyToastInstance`. So I think this PR should be good to go, but 2 follow-up questions:
- The current docs include both the `close()` and `closed` properties. I added the JSDocs comment for `closed`, but `docs-tools` does not create it. Is this correct? I assume so given that `closed` is private, but I just want to make sure.
- The current docs also indicate that both properties are required. I don't think this makes sense and that it's probably an error in the current docs, but I want to confirm that's correct.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720](https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720)

File: src/app/public/modules/toast/toast-instance.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast-instance.ts]: - Did you add the docs above this property? https://github.com/blackbaud/skyux-toast/blob/master/src/app/public/modules/toast/toast-instance.ts#L12.
- Yeah, they shouldn't be required.

Comment (blackbaud-johnly) [reply src/app/public/modules/toast/toast-instance.ts]: Looks like I had the `closed` property JSDocs comment on the wrong line (line 17, which is what this comment thread started on), but when I move it to line 12 as you indicate and serve locally, that property still doesn't show up in the docs:
image
image
Any idea what I'm doing wrong?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720](https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720)

File: src/app/public/modules/toast/toast-instance.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast-instance.ts]: I'll take a look.

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast-instance.ts]: I have a PR in docs-tools that will fix this.

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast-instance.ts]: https://github.com/blackbaud/skyux-docs-tools/pull/102

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/toast/toast-instance.ts]: I upgraded docs-tools and this is working now.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720](https://developer.blackbaud.com/skyux-toast?_sp.spa=pr-set-up-docs-tools-080720)

File: src/app/public/modules/toast/toast-instance.ts (hunk 3 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/toast/toast-instance.ts]: Thanks, Steve.

**Pull request merged**